src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2272,48 +2272,43 @@ int32_t SystemNative_Socket(int32_t addressFamily, int32_t socketType, int32_t p
     return *createdSocket != -1 ? Error_SUCCESS : SystemNative_ConvertErrorPlatformToPal(errno);
 }
 
-int32_t SystemNative_GetAtOutOfBandMark(intptr_t socket, int32_t* atMark)
+
+// Posix defines the prototype for ioctl as ioctl(int, int, ...). 
+// However, Glibc implements it as ioctl(int, unsigned long, ...) and the
+// MUSL library (used by the Alpine OS) implements it as ioctl(int, int, ...)
+#if defined(__GLIBC__) && defined(__GLIBC_MINOR__)
+int32_t GetInt32Ioctl(intptr_t socket, unsigned long request, int32_t* available)
+#else
+int32_t GetInt32Ioctl(intptr_t socket, int request, int32_t* available)
+#endif
 {
-    if (atMark == NULL)
+    if (available == NULL)
     {
         return Error_EFAULT;
     }
 
     int fd = ToFileDescriptor(socket);
 
-    int result;
+    int avail;
     int err;
-    while ((err = ioctl(fd, SIOCATMARK, &result)) < 0 && errno == EINTR);
+    while ((err = ioctl(fd, request, &avail)) < 0 && errno == EINTR);
     if (err == -1)
     {
-        *atMark = 0;
         return SystemNative_ConvertErrorPlatformToPal(errno);
     }
 
-    *atMark = (int32_t)result;
+    *available = (int32_t)avail;
     return Error_SUCCESS;
 }
 
-int32_t SystemNative_GetBytesAvailable(intptr_t socket, int32_t* available)
+int32_t SystemNative_GetAtOutOfBandMark(intptr_t socket, int32_t* available)
```

Comment:
I don't have a strong opinion about whether we take this change or just keep the code as it is and close this PR, with a slight preference for not adding more ifdefs and just closing the PR.  But if we do take the change, we should keep the parameter name as `atMark` rather than `available`.  And the `available` parameter to `GetInt32Ioctl` should also be renamed to `result` or something similar (the `avail` local should also be changed).

Reply:
I'll go with the simpler path of keeping the existing code. I'll close this PR.